tests/test_segment_with_feature.py

```diff
@@ -0,0 +1,1279 @@
+
+    end_indices = non_zero_index[split_indices] + 1
+    # We need to include the last value of the non-zero index
+    # to the end indices, since no split was done there
+    end_indices = [int(end) for end in end_indices]
+    max_non_zero = int(max(non_zero_index) + 1)
+    end_indices.append(max_non_zero)
+    start_indices = [end - length for end, length in zip(end_indices, frame_lengths)]
+    ends = [pd.to_timedelta(end / sampling_rate, unit="s") for end in end_indices]
+    starts = [pd.to_timedelta(start / sampling_rate, "s") for start in start_indices]
+    means = [frame.mean() for frame in frames]
+    index = pd.MultiIndex.from_tuples(zip(starts, ends), names=["start", "end"])
+    return pd.Series(data=means, dtype=signal.dtype, index=index)
+
+
+def segment_with_mean(signal, sampling_rate, *, win_size=1.0, hop_size=1.0):
+    size = signal.shape[1] / sampling_rate
+    starts = pd.to_timedelta(
+        np.arange(0, size - win_size + (1 / sampling_rate), hop_size), unit="s"
+    )
+    ends = pd.to_timedelta(
+        np.arange(win_size, size + (1 / sampling_rate), hop_size), unit="s"
+    )
+    frames = audinterface.utils.sliding_window(
+        signal, sampling_rate, win_size, hop_size
+    )
+    means = frames.mean(axis=(0, 1))
+    index = pd.MultiIndex.from_tuples(zip(starts, ends), names=["start", "end"])
+    return pd.Series(data=means, index=index)
+
+
+def segment_with_mean_std(signal, sampling_rate, *, win_size=1.0, hop_size=1.0):
+    size = signal.shape[1] / sampling_rate
+    starts = pd.to_timedelta(
+        np.arange(0, size - win_size + (1 / sampling_rate), hop_size), unit="s"
+    )
+    ends = pd.to_timedelta(
+        np.arange(win_size, size + (1 / sampling_rate), hop_size), unit="s"
+    )
+    frames = audinterface.utils.sliding_window(
+        signal, sampling_rate, win_size, hop_size
+    )
+    means = frames.mean(axis=(0, 1))
+    stds = frames.std(axis=(0, 1))
+    index = pd.MultiIndex.from_tuples(zip(starts, ends), names=["start", "end"])
+    features = list(np.stack((means, stds), axis=-1))
+    return pd.Series(data=features, index=index)
+
+
+def write_files(paths, signals, sampling_rate):
+    for path, signal in zip(paths, signals):
+        af.write(path, signal, sampling_rate)
+
+
+@pytest.mark.parametrize(
+    "signal, sampling_rate, segment_with_feature, expected",
+    [
+        # invalid multiindex return type
+        pytest.param(
+            ONES_1D,
+            SAMPLING_RATE,
+            audinterface.SegmentWithFeature(
+                feature_names="feature",
+                process_func=lambda x, sr: pd.Series(
+                    data=[1.0],
+                    index=pd.MultiIndex.from_tuples(
+                        [("a", "b")], names=["start", "end"]
+                    ),
+                ),
+            ),
+            None,
+            marks=pytest.mark.xfail(raises=ValueError),
+        ),
+        # incorrectly returning dataframe instead of series
```

Comment:
Could you maybe also add a test what happens if you return only the multiindex instead of a series.